proposals/0207-raise-block-limits-to-50M.md

```diff
@@ -0,0 +1,85 @@
+---
+simd: '0207'
+title: Raise Block Limits to 50M CUs
+authors:
+  - Andrew Fitzgerad (Anza)
+category: Standard
+type: Core
+status: Review
+created: 2024-12-05
+feature:
+supersedes:
+superseded-by:
+extends:
+---
+
+## Summary
+
+Raise the block limit from 48M to 50M CUs.
+
+## Motivation
+
+Current block limits are set to 48M CUs.
+Block limits's primary purpose is to make sure that the vast majority of
+network participants are able to keep up with the network, by restricting the
+amount of work a leader is allowed to pack into a block.
+However, current main net beta traffic is largely not constrained by large
+block execution times.
+This proposal aims a modest increase in block limits to 50M CUs, in order to
+give some additional capacity to the network, and client implementations ready
+for future increases as the performance of the network improves.
+
+## New Terminology
+
+No new terms, but the following definitions are given for clarity:
+
+- Max Block Units - The maximum number of CUs that can be included in a block.
+- Max Writable Account Units - The maximum number of CUs that can be used to
+  write any given account in a block.
+- Max Vote Units - The maximum number of CUs that can be used by simple votes
+  in a block.
+- Max Block Accounts Data Size Delta - The maximum size of the accounts data
+  size delta in a block.
+
+## Detailed Design
+
+The following table shows the current block limits and the proposed block
+limits:
+
+| Type | Current Block Limit | Proposed Block Limit |
+|------|----------------------|----------------------|
+| Max Block Units | 48M | 50M |
+| Max Writable Account Units | 12M | 12M |
+| Max Vote Units | 36M | 36M |
+| Max Block Accounts Data Size Delta | 100MB | 100MB |
+
+## Alternatives Considered
+
+- Leave the block limits as they are
+  - This leaves capacity on the table, we want to get ready for future
+    increases.
+- Double limit to 96M CUs
+  - Viewed as too aggressive at this time, and may cause unforeseen issues
+    particularly in infrastructure supporting the network users.
+  - We instead plan to increase the limits incrementally as the network
```

Comment:
Can we get a more concrete idea of what this looks like? I think doing a small increase first is prudent, but I think the increases proceeding that should be non-linear and increasing in %

Reply:
It's gonna depend how things go, and how quickly we can improve performance.
To the best of my knowledge, we have no plans for some automatic increasing limit at this time; any further changes will go through same SIMD process.

Reply:
Is the feature gate actual required? It would be nice to have automatic progression and so long as you use a static calculation based off of numbers already under consensus (e.g. Epoch or Slot) the validators should be able to easily calculate the current block limit.

Reply:
The feature gate gives us the ability to raise the limit at a time when we feel the cluster is healthy and can handle the additional load that the higher limit could introduce. Adding in "raise limit at epoch N" adds unknowns; what if the cluster is struggling when epoch N comes up or what if the entire validator set hasn't adopted a new enough version by the time epoch N rolls around.

> It would be nice to have automatic progression

This is somewhat similar to what 7layermagik mentioned [here](https://github.com/solana-foundation/solana-improvement-documents/pull/207#issuecomment-2521143758). But to quote Andrew from [here](https://github.com/solana-foundation/solana-improvement-documents/pull/207#issuecomment-2521252932), we are viewing this as a "foot in the door" feature. Implementing this SIMD will establish the logic to raise the limit which will enable future SIMDs to raise the limit to be easier